# GRIP Task-6

### Objective of the task
We have iris flower dataset and we want to classify them as their relevant species.

# Import libraries


In [1]:
# First of all we will import all relevant libraries.
import pandas as pd # for data manipulation
import numpy as np # for data computation
import matplotlib.pyplot as plt #for data visualization
import seaborn as sns
sns.set()
from sklearn.tree import DecisionTreeClassifier # for model building
from sklearn import datasets #for loading datasets
from sklearn.model_selection import train_test_split # for splitting our dataset 
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix # importing matrics for evalution of our result

In [2]:
# Now we will load the data in iris variable
iris = datasets.load_iris()

In [3]:
# X variable is  independent variabe which will be used to predict our target variable
X = pd.DataFrame(iris.data, columns = iris.feature_names)

In [4]:
# y variable is our target variable
y = iris.target

In [5]:
# Shape of independent variable
X.shape

(150, 4)

In [6]:
X.head()

sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)
0                5.1               3.5                1.4               0.2
1                4.9               3.0                1.4               0.2
2                4.7               3.2                1.3               0.2
3                4.6               3.1                1.5               0.2
4                5.0               3.6                1.4               0.2

In our iris dataset we have 150 rows and 4 columns. This dataset has four features such as sepal length, sepal width, petal length, petal width and by using this we will predict  species of iris flower.

In [7]:
# Shape of dependent variable
y.shape

(150,)

Target variable has 150 rows and 1 species column.  

In [8]:
# We are splitting our dataset
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.33,random_state=123)

In [9]:
# Model building
model = DecisionTreeClassifier()

In [10]:
# fitting our model on our training dataset
model.fit(X_train,y_train)

DecisionTreeClassifier()

In [11]:
# Making prediction by using test dataset
pred = model.predict(X_test)

In [12]:
# Printing our accuracy score
print(accuracy_score(y_test, pred))

0.94


In [13]:
# Using classification report we will explore our result. 
print(classification_report(y_test, pred))

              precision    recall  f1-score   support

           0       1.00      1.00      1.00        20
           1       0.79      1.00      0.88        11
           2       1.00      0.84      0.91        19

    accuracy                           0.94        50
   macro avg       0.93      0.95      0.93        50
weighted avg       0.95      0.94      0.94        50



In [14]:
# Printing Confusion Matrix
print(confusion_matrix(y_test, pred))

[[20  0  0]
 [ 0 11  0]
 [ 0  3 16]]
